### DISCUSSION TITLE
Setting the water EOS for reactive transport models

### DISCUSSION THREAD

**Discussion by j-engelmann** [reaktoro/reaktoro#387] [Q&A]
### Description

Hi!

I am in the process of developing reactive fluid flow models on the basis of OpenFOAM, using the mines database through thermofun. So far, i absolutely love the way Reaktoro works in both python and C++, coupling it for some very simple cases was quite smooth. 

I am currently wondering if there is a way to control which EOS for water is used in the solver. Ideally, I would like to make sure that the density of water is calculated with the same equation in both Reaktoro and the transport part, or even better calculate it just once. 

The impact on my current setup would be through the value of `specificVolume()` of the whole system (i create/update a `ChemicalProps` object from my `ChemicalState` after equilibration), which is mostly water. So i guess another question would be how that is calculated and if it takes into account any of the water EOS implemented in ThermoFun.

Thanks in advance for any help!

jasper

### What is your operating system?

macOS

**Comment by allanleal** [comment]
Hi @j-engelmann, thanks for your question and feedback. It is currently not possible to choose which EOS for water is used inside of standard thermodynamic models implemented in Reaktoro. The water EOS used depends on which underlying standard thermodynamic model has been chosen. For example, using any SUPCRT database in Reaktoro implies the use of Wagner & Pruss (2002) water EOS (instead of HGK). If you are using a PHREEQC database, then a simplified equation will be used, to be fully consistent with how PHREEQC calculates logK, its pressure correction, and species standard molar volumes. 

**Note:** Arbitrarily changing the water EOS (or any other model within a larger model) in a standard thermodynamic model may result in previously calibrated parameters no longer providing the same accuracy. This needs to be done with care. 

You are using ThermoFun, I think it is possible to perhaps change the water EOS from WP to that of HGK. @gdmiron - what are your thoughts about this?

**Reply by j-engelmann** [reply]
Hi @allanleal, thanks for the detailed reply! 

Makes sense that it can't be chosen arbitrarily, depending on the underlying model. The MINES database is based on SUPCRT for many species as far as i know. So it should be using Wagner & Pruss then, right? Is there a way to check/find out what it is using?

Part of why i asked like this is because i currently calculate the water properties (density, viscosity, heat capacity etc) needed for OpenFOAM through a different third party library, but would prefer to have it all in one place. Since i need Reaktoro anyways to calculate chemical equilibrium, getting these properties from it or through thermoFun too seems like the logical way forward. Here, knowing which EOS is used would probably help to plan the next steps.

**Reply by allanleal** [reply]
Support for the MINES database in Reaktoro happens via ThermoFun. If I'm not mistaken, by default the Wagner & Pruss (2002) EOS is used. @gdmiron - is this correct?

**Reply by gdmiron** [reply]
Sorry for the delayed reply!

@allanleal The MINES uses WP EOS as default. 

@j-engelmann you can change it to HGK if needed. You need to change the water 'H2O@' record method [code (available options)](https://github.com/thermohub/thermofun/blob/ecf510b841f87f56165eb8d39e484d2d1c2e4728/ThermoFun/GlobalVariables.h#L87) in the database .josn file. You would need to initialize a thermofun database explicitly and then use it to initialize reaktoro.

Go to mines16.json or mines19.json and in the 'H2O@' record and change

```
          "method": {
            "32": "water_eos_iapws95_reaktoro"
          }
```
to 
 ```
          "method": {
            "31": "water_eos_hgk84_reaktoro"
          }
```
or 
 ```
          "method": {
            "29": "water_eos_hgk84_lvs83_gems"
          }
```
- this uses a different EoS (LVS83) close to the critical point of water. 

then initialize the database the file containing the change 

```
# Define the Thermofun database
db = ThermoFunDatabase("<path to>mines19.json")
```
you can find .json ThermoFun databases in Reaktoro resources or [here](https://github.com/thermohub/thermohub/tree/master/thermofun)

**Reply by j-engelmann** [reply]
thanks, i'll keep that in mind for the future!
